### ISSUE TITLE
Installation stuck at "About 2 minutes remaining"

### ISSUE THREAD

**Issue by WhoCannotBeNamed** [foxlet/macOS-Simple-KVM#75]
As the title says, the istallation progress stop at "About 2 minutes remaining". At the moment I have been waiting in front of that screen for over an hour and nothing happens.
![2_minutes](https://user-images.githubusercontent.com/48597596/61212210-dca33480-a701-11e9-8834-de207f2f338d.png)

**Comment by Julioevm** [comment]
I had to leave a machine installing overnight... I'd say be patient.

**Comment by foxlet** [comment]
You may want to switch to tap networking if usermode is slow.

**Comment by freeubi** [comment]
I have the same issue. If i open the installer logs, i can see that the downloaded package could not validated, so its downloading again, and again and again...

**Comment by nextsux** [comment]
Same problem here. It just downloads InsallESDDmg.pkg over and over and every time fails with `xar_verify failed`. I've suspected slow network was the problem but not. I've started local http proxy so file was downloaded almost instantly on second try and result was the same. I've checked checksums, file is downloaded correctly.

I've tested Mojave and HS. Results are the same.

**Comment by freeubi** [comment]
I managed to use a workaround:
- i downloaded the installer on a mac, made a bootable pendrive from that, then made a dmg from the pendrive, converted to img and used that image instead of BaseImage.

Only tested HS now, later i will Catalina too, but i assume it will work too :)

**Comment by nextsux** [comment]
I'm trying to get one right now. Good to know I'm on a right track. Unfortunately I'm without working OSX so preparing it myself is a little problem.

**Comment by freeubi** [comment]
I can share my HS dmg with you.
Send me an email i we can talk the details without spamming this thread :)

**Comment by videah** [comment]
![image](https://user-images.githubusercontent.com/4761786/62108319-ae952700-b2a1-11e9-9cd9-bbe03943dd91.png)

Getting the same issue.

**Comment by nextsux** [comment]
I was investigating issue a little and found that image is downloaded just fine. Image contains sha1 digest and if you try to extract it in VM you'll get an error "Value too large to be stored in data type". It's not clear to me why it's happening though.

**Comment by videah** [comment]
It should be noted that I don't get this issue on every platform. This only seems to happen on my Thinkpad, it installs fine on my Ryzen desktop and my cloud instance.

**Comment by foxlet** [comment]
How big are the target disks images and what networking system is being used (user or tap)? What filesystem is being used (HFS or APFS)?

**Comment by videah** [comment]
32G disk, user networking, and HFS (using APFS makes the install so slow for me that I never reach the 2 minute remaining mark anyway)

**Comment by lucaszanella** [comment]
Same here. Took more than 1 hour to finish. Then stuck at 2 minutes.



Any solutions?

**Comment by lucaszanella** [comment]
Nevermind, by simply waiting a lot, it worked.

**Comment by foxlet** [comment]
Depending on the speed of your underlaying disk, and filesystem overhead, it can take much longer than usual to finish. I would suggest looking at the setup log to see where the most time had taken.

**Comment by zayn666** [comment]
mine is taking a lot too

**Comment by r-a-o** [comment]
Still struggling with this issue, log shows it tried 11 times and failed. 
I have tried High Sierra, Mojave & Catalina, multiple times. Still no luck. :(

**Comment by r-a-o** [comment]
Update: after 4 more retries, it worked on its own.

**Comment by hjames1** [comment]
> Update: after 4 more retries, it worked on its own.

worked with Catalina？

**Comment by r-a-o** [comment]
@hjames1 Yes, but now that MacOS BigSur is public, you should be able to install it via App Store directly in a couple of days.
